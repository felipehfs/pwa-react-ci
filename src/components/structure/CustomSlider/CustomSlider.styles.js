import styled from "styled-components";

export const Slider = styled.div`
  display: flex;
  overflow-x: auto;
  margin-top: 1rem;

  &::-webkit-scrollbar {
    width: 0;
  } 
`;

export const SliderItem = styled.img`
  min-width: 110px;
  height: 100px;
  border-radius: 15px;
  line-height:110px;
  margin-right: 10px;
`;
