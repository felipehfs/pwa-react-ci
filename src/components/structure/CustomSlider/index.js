import React from 'react'
import { Slider, SliderItem } from './CustomSlider.styles'

export default function CustomSlider({
  items
}) {
  return (
    <Slider>
      {
        items.map(item => (
          <SliderItem src={item.cover} key={item.id} />
        ))
      }
    </Slider>
  )
}
