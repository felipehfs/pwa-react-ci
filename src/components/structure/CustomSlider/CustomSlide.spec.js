import { render } from "@testing-library/react";
import CustomSlider from ".";

const arts = [
  { id: 1, cover: "https://randomuser.me/api/portraits/women/73.jpg" },
  { id: 2, cover: "https://randomuser.me/api/portraits/women/12.jpg" },
  { id: 3, cover: "https://randomuser.me/api/portraits/men/39.jpg" },
  { id: 4, cover: "https://randomuser.me/api/portraits/women/66.jpg" },
  { id: 5, cover: "https://randomuser.me/api/portraits/women/55.jpg" },
];

describe('#CustomSlide', () => {

  test('should the ui be the same', () => {
    const { asFragment } = render(<CustomSlider items={arts}/>);

    const html = asFragment();
    expect(html).toMatchSnapshot();
  })
  
});
