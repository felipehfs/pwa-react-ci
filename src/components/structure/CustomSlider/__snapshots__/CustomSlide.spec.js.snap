// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#CustomSlide should the ui be the same 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ dHswJY"
  >
    <img
      class="sc-gsTCUz bxBdaH"
      src="https://randomuser.me/api/portraits/women/73.jpg"
    />
    <img
      class="sc-gsTCUz bxBdaH"
      src="https://randomuser.me/api/portraits/women/12.jpg"
    />
    <img
      class="sc-gsTCUz bxBdaH"
      src="https://randomuser.me/api/portraits/men/39.jpg"
    />
    <img
      class="sc-gsTCUz bxBdaH"
      src="https://randomuser.me/api/portraits/women/66.jpg"
    />
    <img
      class="sc-gsTCUz bxBdaH"
      src="https://randomuser.me/api/portraits/women/55.jpg"
    />
  </div>
</DocumentFragment>
`;
